pkg/crc/preflight/preflight_checks.go

```diff
@@ -0,0 +1,325 @@
+package preflight
+
+import (
+	"encoding/xml"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"os/exec"
+	"os/user"
+	"regexp"
+	"strings"
+
+	units "github.com/docker/go-units"
+)
+
+const (
+	driverBinaryDir      = "/usr/local/bin"
+	kvmDriverBinaryPath  = driverBinaryDir + "/docker-machine-driver-kvm"
+	kvmDriverDownloadURL = "https://github.com/dhiltgen/docker-machine-kvm/releases/download/v0.10.0/docker-machine-driver-kvm-centos7"
+	defaultPoolSize      = "20 GB"
+)
+
+type poolXMLAvailableSpace struct {
+	Available float64 `xml:"available,unit"`
+}
+
+func checkVirtualizationEnabled() (bool, error) {
+	// Check if the cpu flags vmx or svm is present
+	out, err := ioutil.ReadFile("/proc/cpuinfo")
+	if err != nil {
+		return false, err
+	}
+	re := regexp.MustCompile(`flags.*:.*`)
+
+	flags := re.FindString(string(out))
+	if flags == "" {
+		return false, errors.New("Could not find cpu flags from /proc/cpuinfo")
+	}
+
+	re = regexp.MustCompile(`(vmx|svm)`)
+
+	cputype := re.FindString(flags)
+	if cputype == "" {
+		return false, errors.New("Virtualization is not available for you CPU")
+	}
+	return true, nil
+}
+
+func fixVirtualizationEnabled() (bool, error) {
+	return false, errors.New("You need to enable virtualization in bios")
+}
+
+func checkKvmEnabled() (bool, error) {
+	// Check if /dev/kvm exists
+	if _, err := os.Stat("/dev/kvm"); os.IsNotExist(err) {
+		return false, errors.New("kvm kernel module is not loaded")
+	}
+	return true, nil
+}
+
+func fixKvmEnabled() (bool, error) {
+	cmd := exec.Command("sudo", "modprobe", "kvm")
+	err := cmd.Run()
+	if err != nil {
+		return false, err
+	}
+	return true, nil
+}
+
+func checkLibvirtInstalled() (bool, error) {
+	path, err := exec.LookPath("virsh")
+	if err != nil {
+		return false, errors.New("Libvirt cli virsh was not found in path")
+	}
+	fi, _ := os.Stat(path)
+	if fi.Mode()&os.ModeSymlink != 0 {
+		path, err = os.Readlink(path)
+		if err != nil {
+			return false, errors.New("Libvirt cli virsh was not found in path")
+		}
+	}
+	return true, nil
+}
+
+func fixLibvirtInstalled() (bool, error) {
+	cmd := exec.Command("sudo", "dnf", "install", "-y", "libvirt", "libvirt-devel", "libvirt-daemon-kvm", "qemu-kvm")
```

Comment:
why `libvirt-devel`?

Reply:
IIRC I saw this requirement in the installer readme and added it here.. i'll recheck and verfiy